pkg/health/server/server.go

```diff
@@ -264,14 +279,49 @@ func (s *Server) collectNodeConnectivityMetrics() {
 				location, metrics.LabelPeerNode, metrics.LabelTrafficICMP, metrics.LabelAddressTypeSecondary)
 		}
 	}
+
+	// Aggregated health statuses for endpoint connectivity
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerEndpoint, metrics.LabelReachable).
+		Set(float64(endpointStatuses[healthClientPkg.ConnStatusReachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerEndpoint, metrics.LabelUnreachable).
+		Set(float64(endpointStatuses[healthClientPkg.ConnStatusUnreachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerEndpoint, metrics.LabelUnknown).
+		Set(float64(endpointStatuses[healthClientPkg.ConnStatusUnknown]))
+
+	// Aggregated health statuses for node connectivity
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerNode, metrics.LabelReachable).
+		Set(float64(nodeStatuses[healthClientPkg.ConnStatusReachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerNode, metrics.LabelUnreachable).
+		Set(float64(nodeStatuses[healthClientPkg.ConnStatusUnreachable]))
+
+	metrics.NodeHealthConnectivityStatus.WithLabelValues(
+		localClusterName, localNodeName, metrics.LabelPeerNode, metrics.LabelUnknown).
+		Set(float64(nodeStatuses[healthClientPkg.ConnStatusUnknown]))
 }
 
 func collectConnectivityMetric(status *healthModels.ConnectivityStatus, labels ...string) {
-	var metricValue float64 = -1
+	healthLabels := make([]string, 5)
+	copy(healthLabels, labels[0:2])
+	copy(healthLabels[2:], labels[6:])
 	if status != nil {
-		metricValue = float64(status.Latency) / float64(time.Second)
+		if status.Status == "" {
+			metricValue := float64(status.Latency) / float64(time.Second)
+			metrics.NodeConnectivityLatency.WithLabelValues(labels...).Set(metricValue)
+			metrics.NodeHealthConnectivityLatency.WithLabelValues(healthLabels...).Observe(metricValue)
+		} else {
+			metrics.NodeHealthConnectivityLatency.WithLabelValues(healthLabels...).Observe(60)
```

Comment:
Where does this 60 come from, is this a default timeout? If so maybe can we create a constant.

Reply:
Yes, default timeout. I will use a constant, thank you for the pointer